### PR TITLE
fix: New route for externals

### PR DESCRIPTION
**Pull request by Lnamw**

### CODE REVIEW

**Comment by Lnamw** [comment]
En attente du changement de la réponse coté back pour pouvoir tester

File: MailCore/API/Endpoint.swift [hunk 1]
```diff
@@ -190,6 +190,10 @@ public extension Endpoint {
         ])
     }
 
+    static func externalMailFlag(hostingId: Int, mailboxName: String) -> Endpoint {
+        return .baseManager.appending(path: "/\(hostingId)/mailboxes/\(mailboxName)/external_mail_flag")
```
Comment (valentinperignon) [inline MailCore/API/Endpoint.swift:194]: We now have multiple endpoints starting with `.baseManager.appending(path: "/\(hostingId)/mailboxes/\(mailboxName)/`, would it be possible to refactor?

File: MailCore/API/MailApiFetcher/MailApiFetcher+Common.swift [hunk 2]
```diff
@@ -98,6 +98,14 @@ public extension MailApiFetcher {
         try await perform(request: authenticatedRequest(.quotas(mailbox: mailbox.mailbox, productId: mailbox.hostingId)))
     }
 
+    func externalMailFlag(mailbox: Mailbox) async throws -> ExternalMailInfo {
+        try await perform(request: authenticatedRequest(.externalMailFlag(
+            hostingId: mailbox.hostingId,
+            mailboxName: mailbox.mailbox
+        )))
+        .data
```
Comment (valentinperignon) [inline MailCore/API/MailApiFetcher/MailApiFetcher+Common.swift:106]: We don't need `.data` anymore, you can safely remove it 😊

It’s thanks to a recent PR from Philippe on Core ! If you want to take a look, it's the function `perform<T: Decodable>(request: DataRequest, decoder: JSONDecoder = ApiFetcher.decoder) async throws -> T` on `InfomaniakCore.ApiFetcher.swift:153`

File: MailCore/Cache/AccountManager/AccountManager.swift [hunk 3]
```diff
@@ -259,6 +259,7 @@ public final class AccountManager: RefreshTokenDelegate, ObservableObject {
             if mailbox.isLimited {
                 mailbox.quotas = try await apiFetcher.quotas(mailbox: mailbox)
             }
+            mailbox.externalMailInfo = try await apiFetcher.externalMailFlag(mailbox: mailbox)
```
Comment (valentinperignon) [inline MailCore/Cache/AccountManager/AccountManager.swift:262]: As we have 3 calls to await here (permissions, quotas and external), would it be possible to use the `async let` syntax?
We can try together if you want!

**Review by valentinperignon — COMMENTED** [review]
👍

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**